src/parser_utils.c

```diff
@@ -102,6 +109,34 @@ t_err	ft_create_str_arr(t_cmd *tmp, int count_arg, int count_delim)
 	}
 	else
 		tmp->delims = NULL;
+	err = ft_create_str_arr2(tmp, count_out);
+	return (err);
+}
+
+/**
+ * @brief Reserve memory space for outfiles string array.
+ * 
+ * @param tmp 		New cmd struct to be created.
+ * @param count_out Amount of outfiles.
+ * @return t_err 	ERR_MALLOC, SUCCESS
+ */
+t_err	ft_create_str_arr2(t_cmd *tmp, int count_out)
+{
+	if (count_out)
+	{
+		tmp->outfiles = malloc(sizeof(char *) * (count_out + 1));
+		if (!tmp->outfiles)
+			return (ERR_MALLOC);
+		tmp->outfiles[count_out] = NULL;
+		tmp->append_switches = malloc(sizeof(bool) * count_out);
```

Comment:
Needs to free tmp->outfiles if fails

Reply:
General cleanup will free all malloced variables